pkg/common/plugin/azure/msi.go

```diff
@@ -35,16 +41,8 @@ type MSIAttestationData struct {
 
 type MSITokenClaims struct {
 	jwt.Claims
-	TenantID string `json:"tid,omitempty"`
-}
-
-func (c *MSITokenClaims) AgentID(trustDomain string) string {
-	u := url.URL{
-		Scheme: "spiffe",
-		Host:   trustDomain,
-		Path:   path.Join("spire", "agent", "azure_msi", c.TenantID, c.Subject),
-	}
-	return u.String()
+	TenantID    string `json:"tid,omitempty"`
+	PrincipalID string `json:"sub,omitempty"`
```

Comment:
I did a double take and had to remind myself what happens with the JSON decoder when there are two fields with the same tag. Fortunately the top level field takes precedence over the embedded struct but this is going to be a little subtle if there is any code that tries to use the `.Subject` field (since it will be unset). Should be easy enough to catch in testing though. I think we're ok here.

Reply:
I thought about creating a new struct for only storing claims/fields that we want to expose and use. But we would still have another struct being composed by  jwt.Claims only, since we use the `ValidateWithLeeway` method for validation, but I'm not sure if that would make the code clearer since we will be going to deserialize the token two times for two different structs, what do you think?

Reply:
I think we're ok with it as-is. If the `Subject` field is used at all the unit-tests should catch that it is empty.